tt-train/sources/ttml/optimizers/sgd.cpp

```diff
@@ -47,6 +52,26 @@ void SGD::step() {
         }
 
         auto gradients = tensor_ptr->get_grad();
+        if (devices > 1U) {
```

Comment:
I'd put it into the separate function and call it in sgd, adamw and etc.

Reply:
Will do, makes a lot of sense!